aws/lambda/pytorch-auto-revert/pytorch_auto_revert/workflow_resolver.py

```diff
@@ -102,3 +118,87 @@ def _build_indices(self) -> None:
                     ref = WorkflowRef(display_name=name, file_name=base)
                     self._by_display[name] = ref
                     self._by_file[base] = ref
+
+    def get_input_support(self, workflow_name: str) -> WorkflowInputSupport:
+        """Check if workflow supports filtering inputs by parsing its YAML.
+
+        Args:
+            workflow_name: Display name or file name of the workflow
+
+        Returns:
+            WorkflowInputSupport describing which inputs are accepted
+        """
+        ref = self.require(workflow_name)
+
+        if ref.file_name in self._input_support_cache:
+            return self._input_support_cache[ref.file_name]
+
+        support = self._fetch_and_parse_workflow_inputs(ref.file_name)
+        self._input_support_cache[ref.file_name] = support
+        return support
+
+    def _fetch_and_parse_workflow_inputs(self, file_name: str) -> WorkflowInputSupport:
+        """Fetch workflow YAML from GitHub and parse for dispatch inputs.
+
+        Args:
+            file_name: Workflow file basename (e.g., "trunk.yml")
+
+        Returns:
+            WorkflowInputSupport with detected input support
+        """
+        path = f".github/workflows/{file_name}"
+
+        for attempt in RetryWithBackoff():
+            with attempt:
+                contents = self._repository.get_contents(path)
+                yaml_content = contents.decoded_content.decode("utf-8")
```

Comment:
call `workflow = yaml.safe_load(yaml_content)` here.

you can't trust that the success on do the request will actually lead to a successful response (even if it is 200, etc).

You can get garbage data, so, you might want to retry at this point.